### ISSUE TITLE
Add user auth to the site

### ISSUE THREAD

**Issue by crutchcorn** [unicorn-utterances/unicorn-utterances#71]
One of our long-term goals as a team is to enable users to have a more custom tailored experience in learning CS content. Adding user authentication will enable a lot of functionality that aligns well with our interactive user-driven dream

We'll need to figure out:
- [ ] What do we want/need for privacy sakes?
    - [ ] Have an explicit clear verbose outline that complies with GDPR and is well understood
- [ ] What security do we want/need to have to keep our user data safe?
    - [ ] How do we handle security info issues?
- [ ] What type of initial user portal do we want to have
    - [ ] Design
    - [ ] Functionality
- [ ] How do we migrate current users?
- [ ] Should we allow OAuth?
    - [ ] Should this OAuth data include profile pictures/etc that we should use as user data?
- [ ] How do we want to build out the server?
    - [ ] What language and stack do we want for the server?
    - [ ] Do we want NoSQL or SQL?
    - [ ] Where/how do we want to deploy?

We'll need to break out the tickets when we've had further discussion on this ticket, feedback encouraged

Blocked by: #70

**Comment by crutchcorn** [comment]
Closing this ticket as "Out-of-scope" for the present. We may revisit this in the future.

Realistically, when things like this start getting called into question, it makes more sense to write the site from scratch as a generic product and make UU a consumer of that product than trying to shoe-horn a home-grown CMS into a service-based website
